libkbfs/prefetcher.go

```diff
@@ -0,0 +1,168 @@
+package libkbfs
+
+import (
+	"io"
+	"sort"
+	"sync"
+
+	"golang.org/x/net/context"
+)
+
+const (
+	defaultIndirectPointerPrefetchCount int = 20
+	fileIndirectBlockPrefetchPriority   int = -100
+	dirEntryPrefetchPriority            int = -200
+)
+
+type prefetchRequest struct {
+	priority int
+	kmd      KeyMetadata
+	ptr      BlockPointer
+	block    Block
+}
+
+var _ Prefetcher = (*blockPrefetcher)(nil)
+
+type blockPrefetcher struct {
+	retriever  blockRetriever
+	progressCh chan prefetchRequest
+	shutdownCh chan struct{}
+	doneCh     chan struct{}
+	sg         sync.WaitGroup
+}
+
+func newBlockPrefetcher(retriever blockRetriever) *blockPrefetcher {
+	p := &blockPrefetcher{
+		retriever:  retriever,
+		progressCh: make(chan prefetchRequest),
```

Comment:
Can you add this as a comment for `progressCh`? May as well add comments for `shutdownCh` and `doneCh`, too